zarr/core.py

```diff
@@ -445,21 +460,36 @@ def __getitem__(self, item):
 
         """
 
+        if len(self._shape) == 0:
+            return self._get_basic_selection_zd(selection)
+
+        elif len(self._shape) == 1:
+            # safe to do "fancy" indexing, no ambiguity
+            return self.get_orthogonal_selection(selection)
```

Comment:
You can do vectorized indexing on 1D arrays, too, e.g.,
```
In [22]: a = np.arange(4)

In [23]: a[a.reshape(2, 2)]
Out[23]:
array([[0, 1],
       [2, 3]])
```

More generally, I agree that it's unambiguous for 1D, but given the focus of zarr on N-dimensions I would be reluctant to add this shortcut. The special case feels like more trouble than it's worth.

Reply:
Thanks, yep I think your probably right. I've added support for vectorized indexing with multi-dimensional coordinate arrays, but have limited ``__getitem__`` to basic selections only.